docs/asciidoc/static/complex-pipeline.asciidoc

```diff
@@ -0,0 +1,363 @@
+==== The Logstash Forwarder
+The Logstash Forwarder is a software client that processes data at the machine of origin before sending the data to the 
+Logstash instance. Data from the Logstash Forwarder uses a secure protocol called _lumberjack_ that is designed for 
+reliability and low latency. Default Logstash configuration includes the lumberjack input plugin, which is designed to 
+be resource-friendly. To install the Logstash Forwarder on your data source machine, install the appropriate package 
+from the main Logstash https://www.elastic.co/downloads/logstash[product page].
+
+Create a configuration file for the Logstash Forwarder similar to the following example:
+
+[source,json]
+{
+  "network": {
+    "servers": [ "localhost:5043" ],
+
+    "ssl ca": "/path/to/localhost.crt",
+
+    "timeout": 15
+  },
+
+  "files": [
+    {
+      "paths": [
+        "/path/to/sample-log"
+      ],
+      "fields": { "type": "apache" }
+    }
+  ]
+}
+
+Save this configuration file as `logstash-forwarder.conf`. 
+
+Configure your Logstash instance to use the Lumberjack input plugin by adding the following lines to the `input` section 
+of the `first-pipeline.conf` file:
+
+[source,json]
+lumberjack {
+    port => “5043”
+    ssl_certificate => “/path/to/ssl-cert”
+    ssl_key => “/path/to/ssl-key”
+}
+
+[float]
+[[logstash-file-output]]
+==== Writing Logstash Data to a File
+
+You can configure your Logstash pipeline to write data directly to a file with the 
+{logstash}plugins-outputs-file.html[`file`] output plugin.
+
+Configure your Logstash instance to use the `file` output plugin by adding the following lines to the `output` section 
+of the `first-pipeline.conf` file:
+
+[source,json]
+file {
+    path => /path/to/target/file
+}
+
+[float]
+[[multiple-es-nodes]]
+==== Writing to multiple Elasticsearch nodes
+
+Writing to multiple Elasticsearch nodes lightens the resource demands on a given Elasticsearch node, as well as 
+providing redundant points of entry into the cluster when a particular node is unavailable.
+
+To configure your Logstash instance to write to multiple Elasticsearch nodes, edit the output section of the `first-pipeline.conf` file to read:
+
+[source,json]
+output {
+       elasticsearch {
+                      protocol => “http”
+                      host => [“IP Address 1”, “IP Address 2”, “IP Address 3”]
+        }
+}
+
+Use the IP addresses of three non-master nodes in your Elasticsearch cluster in the host line. 
+
+[float]
+[[testing-second-pipeline]]
+==== Testing the Pipeline
+
+At the data source machine, run the Logstash Forwarder with the following command:
+
+[source,shell]
+logstash-forwarder -config logstash-forwarder.conf
+
+Logstash Forwarder will attempt to connect on port 5403. Until Logstash starts with an active Lumberjack plugin, there 
+won’t be any answer on that port, so any messages you see regarding failure to connect on that port are normal for now.
+
+Start Logstash with the following command:
```

Comment:
How about a "Putting it all together" config for users which combines all these different parts we are talking about